uh_internal/apps/dailyduties/pyexchange.py

```diff
@@ -0,0 +1,38 @@
+import os
+from os.path import join, dirname
+from ...settings.base import get_env_variable
+
+# https://github.com/fedorareis/pyexchange This is a combination of a few branches and some custom code
+from pyexchange import Exchange2010Service, ExchangeNTLMAuthConnection, ExchangeBasicAuthConnection
+
+URL = get_env_variable('RESNET_INTERNAL_OUTLOOK_URL')
+USERNAME = get_env_variable('RESNET_INTERNAL_EMAIL_OUT_USERNAME')  # ResNet username
+PASSWORD = get_env_variable('RESNET_INTERNAL_EMAIL_OUT_PASSWORD')  # ResNet Password
+
+
+def setup():
+    """ Creates the Exchange Connection """
+    # Set up the connection to Exchange
+    connection = ExchangeBasicAuthConnection(url=URL,
+                                             username=USERNAME,
+                                             password=PASSWORD)
+
+    service = Exchange2010Service(connection)
+
+    return service
+
+
+def get_mail(service):
```

Comment:
for efficiency reason lets change the body of this function to be 

``` Pyhon
folder = service.folder()
email = folder.get_folder("inbox")

return email.total_count
```

The method you are currently using gets the xml for all of the emails then parses it, this is a lot more intensive than getting the count from the folder object.